LoRaEngine/test/LoRaWan.IntegrationTest/C2DMessageTest.cs

```diff
@@ -245,5 +245,222 @@ public async Task Test_OTAA_Unconfirmed_Receives_C2D_Message()
             Assert.True(foundReceivePacket, $"Could not find lora receiving message '{expectedRxSerial}'");
                 
         }
+
+        // Ensures that C2D messages are received when working with unconfirmed messages
+        // Uses Device10_OTAA
+        [Fact]
+        public async Task Test_OTAA_Unconfirmed_Receives_Confirmed_FPort_Message()
+        {
+            var device = this.TestFixture.Device15_OTAA;
+            Log($"Starting {nameof(Test_OTAA_Unconfirmed_Receives_Confirmed_C2D_Message)} using device {device.DeviceID}");
+
+            await this.ArduinoDevice.setDeviceModeAsync(LoRaArduinoSerial._device_mode_t.LWOTAA);
+            await this.ArduinoDevice.setIdAsync(device.DevAddr, device.DeviceID, device.AppEUI);
+            await this.ArduinoDevice.setKeyAsync(device.NwkSKey, device.AppSKey, device.AppKey);
+
+            await this.ArduinoDevice.SetupLora(this.TestFixture.Configuration.LoraRegion);
+
+            var joinSucceeded = await this.ArduinoDevice.setOTAAJoinAsyncWithRetry(LoRaArduinoSerial._otaa_join_cmd_t.JOIN, 20000, 5);
+
+            if (!joinSucceeded)
```

Comment:
The code you copied from me was sub-optimal. We don't need an if here. We can just do the Assert.True(...)